### PR TITLE
test/old: force "-i viminfo" in $RUN_VIMTEST

### PR DESCRIPTION
**Pull request by justinmk**
Before this change, some tests fail if the user has a local
shada/viminfo in the default $HOME location. For example:

    TEST_FILE=test_marks.res make oldtest

Ostensibly this change should not be necessary, because of this
statement in runtest.vim:

    set viminfo+=nviminfo

### CODE REVIEW

File: src/nvim/testdir/Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ NEW_TESTS_IGNORE := $(NEW_TESTS_IN_ALOT) $(NEW_TESTS_ALOT) \
   test_listlbr \
   test_largefile \
 
-NEW_TESTS = $(addsuffix .res,$(sort $(filter-out $(NEW_TESTS_IGNORE),$(basename $(notdir $(wildcard test_*.vim))))) $(NEW_TESTS_ALOT))
+NEW_TESTS ?= $(addsuffix .res,$(sort $(filter-out $(NEW_TESTS_IGNORE),$(basename $(notdir $(wildcard test_*.vim))))) $(NEW_TESTS_ALOT))
```
Comment (justinmk) [inline src/nvim/testdir/Makefile]: Unrelated change. @blueyed did you remove the `?=` intentionally? The top-level `Makefile` sets `NEW_TESTS` to allow running a single test like this:

    TEST_FILE=test_marks.res make oldtest

Comment (blueyed) [reply src/nvim/testdir/Makefile]: @justinmk 
No, wasn't intentional I think.
